### ISSUE TITLE
coqc -time should report each character range only once

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#14475]
#### Description of the problem

Consider this file
```coq
Lemma foo : True.
Proof.
  Open Scope nat_scope.
  exact I.
  Close Scope nat_scope.
Qed.
```
`coqc -time` gives
```
Chars 0 - 17 [Lemma~foo~:~True.] 0. secs (0.u,0.s)
Chars 18 - 24 [Proof.] 0. secs (0.u,0.s)
Chars 27 - 48 [Open~Scope~nat_scope.] 0. secs (0.u,0.s)
Chars 51 - 59 [exact~I.] 0. secs (0.u,0.s)
Chars 62 - 84 [Close~Scope~nat_scope.] 0. secs (0.u,0.s)
Chars 27 - 48 [Open~Scope~nat_scope.] 0. secs (0.u,0.s)
Chars 62 - 84 [Close~Scope~nat_scope.] 0. secs (0.u,0.s)
Chars 85 - 89 [Qed.] 0. secs (0.u,0.s)
```

Presumably, the vernacular commands are re-run at Qed-time to handle async proof processing stuff (cc @ejgallego ?)  They should not emit timing info, though.

#### Coq Version

8.13

**Comment by JasonGross** [comment]
Note that this causes an issue with the bug minimizer (https://github.com/JasonGross/coq-tools/issues/63), which relies on `-time` for parsing.  I can work around this issue, but it would be nice to have it fixed, too.

**Comment by SkySkimmer** [comment]
Since they're run twice what's wrong with printing twice?

**Comment by JasonGross** [comment]
Said another way, I think that replaying the global side effects should be considered part of "running the command Qed", rather than the `Qed` vernacular being secretly syntactic sugar for rerunning all the vernaculars of the proof and also saving the proof.
